velox/common/memory/MemoryUsageTracker.cpp

```diff
@@ -31,4 +31,49 @@ std::shared_ptr<MemoryUsageTracker> MemoryUsageTracker::create(
 
   return std::make_shared<SharedMemoryUsageTracker>(parent, type, config);
 }
+
+void MemoryUsageTracker::updateInternal(UsageType type, int64_t size) {
+  // Update parent first. If one of the ancestor's limits are exceeded, it
+  // will throw VeloxMemoryCapExceeded exception.
+  if (parent_) {
+    parent_->updateInternal(type, size);
+  }
+
+  auto newPeak = currentUsageInBytes_[static_cast<int>(type)].fetch_add(
+                     size, std::memory_order_relaxed) +
+      size;
+
+  if (size > 0) {
+    ++numAllocs_[static_cast<int>(type)];
```

Comment:
static_cast<int>(type) repeats a lot making this code harder to read; perhaps, introduce a temporary variable or change method's signature